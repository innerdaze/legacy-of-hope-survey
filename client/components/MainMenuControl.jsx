import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from 'grommet/components/Button'
import MenuIcon from 'grommet/components/icons/base/Menu'

class MainMenuControl extends Component {
  constructor(props) {
    super(props)

    this.showMenu = this.showMenu.bind(this)
  }

  showMenu() {
    this.props.showMenu()
  }

  render() {
    return (
      <Button
        plain
        icon={<MenuIcon/>}
        onClick={this.showMenu}
        />
    )
  }
}

MainMenuControl.propTypes = {
  showMenu: PropTypes.func.isRequired
}

MainMenuControl.defaultProps = {
  showMenu: Function.prototype
}

export default MainMenuControl
