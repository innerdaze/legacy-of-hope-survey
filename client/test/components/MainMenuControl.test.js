import React from 'react'
import MainMenuControl from '../../components/MainMenuControl'

let wrapper, showMenu

describe('Testing on MainMenuControl...', () => {
  beforeEach(() => {
    showMenu = jest.fn()
    wrapper = shallow(<MainMenuControl showMenu={showMenu} />)
  })

  describe('MainMenuControl layout', () => {
    test('Expect MainMenuControl to have a Button', () => {
      expect(wrapper.find('Button')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect showMenu prop function to call if Button click', () => {
      wrapper.find('Button').simulate('click')

      expect(showMenu).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
