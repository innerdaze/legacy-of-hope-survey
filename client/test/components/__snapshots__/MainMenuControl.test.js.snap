// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on MainMenuControl... MainMenuControl layout Expect MainMenuControl to have a Button 1`] = `
<Button
  icon={
    <Menu
      responsive={true}
    />
  }
  method="push"
  onClick={[Function]}
  plain={true}
  type="button"
/>
`;

exports[`Testing on MainMenuControl... Test on prop Expect showMenu prop function to call if Button click 1`] = `
<Button
  icon={
    <Menu
      responsive={true}
    />
  }
  method="push"
  onClick={[Function]}
  plain={true}
  type="button"
/>
`;
